### ISSUE TITLE
Excuse me, is it working well in php 7?

### ISSUE THREAD

**Issue by kn007** [igbinary/igbinary7#3]
It's compile successful in php 7.0.1.

~~But when i using php-memcached with igbinary, segmentation fault.~~( https://github.com/php-memcached-dev/php-memcached/issues/208 )
 **Segmentation fault fixed in php-memcached-dev php7 branch**

When "session.serialize_handler=igbinary" in php.ini.

return error like:

```
[01-Feb-2016 23:55:45]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:51]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:52]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:53]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
```

same return with phpmyadmin https://github.com/phpmyadmin/phpmyadmin/issues/11781

Thank you.

**Comment by kn007** [comment]
log

```
Dec 27 23:38:25 kn007 kernel: php-fpm[21765]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:38:27 kn007 kernel: php-fpm[21778]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:06 kn007 kernel: php-fpm[21807]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:10 kn007 kernel: php-fpm[21793]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:12 kn007 kernel: php-fpm[21758]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:13 kn007 kernel: php-fpm[21794]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:28 kn007 kernel: php-fpm[21775]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:29 kn007 kernel: php-fpm[21938]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:52 kn007 kernel: php-fpm[21777]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:39:52 kn007 kernel: php-fpm[21790]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:40:06 kn007 kernel: php-fpm[21799]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:40:08 kn007 kernel: php-fpm[22791]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
Dec 27 23:40:14 kn007 kernel: php-fpm[21797]: segfault at 10 ip 00007f3fa75dc060 sp 00007ffee413d220 error 6 in igbinary
.so[7f3fa75da000+a000]
```

core dump

```
...
Core was generated by `php-fpm: pool www                                                             '.
Program terminated with signal 11, Segmentation fault.
#0  0x00007f10e829f060 in igbinary_serialize_ex (ret=0x7ffe09d032c0, ret_len=0x10, z=0x7f10ed818db0,
    memory_manager=0x0) at /data/src/igbinary7/igbinary.c:433
433             *ret_len = igsd.buffer_size - 1;
Missing separate debuginfos, use: debuginfo-install bzip2-libs-1.0.5-7.el6_0.x86_64 cyrus-sasl-lib-2.1.23-15.el6_6.2.x86
_64 glibc-2.12-1.166.el6_7.3.x86_64 libgcc-4.4.7-16.el6.x86_64 libidn-1.18-2.el6.x86_64 libmcrypt-2.5.8-9.el6.x86_64 lib
png-1.2.49-2.el6_7.x86_64 libstdc++-4.4.7-16.el6.x86_64 nspr-4.10.8-2.el6_7.x86_64 nss-3.19.1-5.el6_7.x86_64 nss-softokn
-freebl-3.14.3-23.el6_7.x86_64 nss-util-3.19.1-2.el6_7.x86_64 openldap-2.4.40-7.el6_7.x86_64 zlib-1.2.3-29.el6.x86_64
```

**Comment by ibobo** [comment]
From my tests this is working when used directly from php (`igbinary_serialize` and `igbinary_unserialize`) but completely broken when referred by other plugins (like memcached and apcu) and when used as session serialization.

**Comment by kn007** [comment]
@ibobo Same. 

BTW, it's work fine with phpRedis plugin.

**Comment by GuillaumeRossolini** [comment]
I have a similar error:
https://github.com/igbinary/igbinary/issues/58
(sorry for posting in the wrong branch)

**Comment by kn007** [comment]
Segmentation fault fixed in php-memcached-dev php7 branch, but 

```
session.serialize_handler=igbinary
```

still return error like:

```
[01-Feb-2016 23:55:45]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:51]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:52]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
[01-Feb-2016 23:55:53]PHP Warning:  session_start(): Cannot find serialization handle
r 'igbinary' - session startup failed in ~WWW/album/imagevue.php on line 5"
```

**Comment by SquallATF** [comment]
@kn007 session  serialize handler  is disabled on php7
https://github.com/igbinary/igbinary7/blob/master/igbinary.c#L25

**Comment by kn007** [comment]
@SquallATF THX, but i try msgpack module early on php7, it,s ok.
You mean, it's disabled handler on php7 or on igbinary php7 branch?

@laruence Is disabled session serialize handler on php7?

**Comment by SquallATF** [comment]
@kn007 on igbinary php7 branch

**Comment by kn007** [comment]
@SquallATF Thank you.

Any idea to fix it? Or just do not using igbinary to handle session serialize?

**Comment by SquallATF** [comment]
@kn007  Just do not using igbinary to handle session serialize or FIX The code https://github.com/igbinary/igbinary7/blob/master/igbinary.c#L25

**Comment by kn007** [comment]
@SquallATF OK, Thanks.

**Comment by TysonAndre** [comment]
The session serialization/unserialization handler was re-enabled in July.

Also, after this issue was fixed in #20, development was moved to the https://github.com/igbinary/igbinary gbinary/igbinary repository(src/php7). https://github.com/igbinary/igbinary/releases/tag/2.0.0
